src/Plugin/Satis/UpdateAndBuildJob.php

```diff
@@ -9,20 +9,87 @@
 
 namespace Terramar\Packages\Plugin\Satis;
 
+use Doctrine\ORM\EntityManager;
+use Symfony\Component\Filesystem\Filesystem;
 use Symfony\Component\Process\PhpExecutableFinder;
 use Symfony\Component\Process\ProcessBuilder;
+use Terramar\Packages\Entity\Package;
 use Terramar\Packages\Job\ContainerAwareJob;
 
 class UpdateAndBuildJob extends ContainerAwareJob
 {
+    /**
+     * @return string
+     */
+    private function getCacheDir(Package $package)
+    {
+        return $this->getContainer()->getParameter('app.cache_dir') . '/auth/' . $package->getFqn();
+    }
+
+    /**
+     * @return EntityManager
+     */
+    private function getEntityManager()
+    {
+        return $this->getContainer()->get('doctrine.orm.entity_manager');
+    }
+
     public function run($args)
     {
+        /** @var Package $package */
+        $package = $this->getEntityManager()->getRepository(Package::class)->find($args['package_id']);
+
+        $config = [];
+        $remote = $package->getRemote();
+        switch ($remote->getAdapter()) {
+            case 'GitHub':
+                /** @var \Terramar\Packages\Plugin\GitHub\RemoteConfiguration $remoteConfig */
+                $remoteConfig = $this->getEntityManager()->getRepository('Terramar\Packages\Plugin\GitHub\RemoteConfiguration')
+                    ->findOneBy(['remote' => $remote]);
+                if ( ! $remoteConfig) {
+                    throw new \RuntimeException('Unable to find RemoteConfiguration for ' . $remote->getAdapter() . ' ' . $remote->getName());
+                }
+
+                $config['github-oauth']['github.com'] = $remoteConfig->getToken();
+
+                break;
+
+            case 'GitLab':
+                /** @var \Terramar\Packages\Plugin\GitLab\RemoteConfiguration $remoteConfig */
+                $remoteConfig = $this->getEntityManager()->getRepository('Terramar\Packages\Plugin\GitLab\RemoteConfiguration')
+                    ->findOneBy(['remote' => $remote]);
+                if ( ! $remoteConfig) {
+                    throw new \RuntimeException('Unable to find RemoteConfiguration for ' . $remote->getAdapter() . ' ' . $remote->getName());
+                }
+
+                $config['gitlab-token']['gitlab.com'] = $remoteConfig->getToken();
```

Comment:
This should use the base hostname from the $remoteConfig instead of `gitlab.com` for the key. However, that still doesn't fix (for either GitHub or GitLab) supporting multiple tokens for one instance of a host, which is something at least me needs.

Reply:
Changed. Since the `UpdateAndBuildJob` now runs for only one repository at a time, and creates a specific `auth.json` for each it should overwrite this with whatever gitlab/github instance you have for that one repo.

Reply:
Ah, good call! :+1: